flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherResourceCleanupTest.java

```diff
@@ -762,4 +850,29 @@ public JobManagerRunner createJobManagerRunner(
             throw testException;
         }
     }
+
+    private static class TestingHistoryServerArchivist implements HistoryServerArchivist {
```

Comment:
This class is kind of obsolete. We can replace any of its occurrences with a simple callback `executionGraphInfo -> new CompletedFuture()` (for never completing archiving) or `executionGraphInfo -> future` if you want to provide the future in the test method itself to simulate the completion of the archiving. WDYT?